test/contrib/flag/test_flag.py

```diff
@@ -107,3 +108,58 @@ def test_zero_grad(model: torch.nn.Module, optimizer: torch.nn.Module,
         for p in group['params']:
             assert torch.count_nonzero(
                 p.grad) == 0, "After running FLAG, the gradients should be 0!"
+
+
+class FLAGLossHistoryCallback(FLAGCallback):
+  def __init__(self):
+    self.loss_history = []
+
+  def on_ascent_step_end(self, step_index, loss, perturb):
```

Comment:
nit: the argument here is `perturb`, but the argument in `FLAGCallback` is `perturb_data`?